### PR TITLE
runtime/libia2: Implement __wrap_main, exit and allocate thread-local stacks

### PR DESCRIPTION
**Pull request by ayrtonm**
Still need to figure out x18 handling and check the ARM64 ABI spec for stack alignment requirements.

### CODE REVIEW

**Comment by ayrtonm** [comment]
ARM64 test was previously failing because we were switching to ia2_stackptr_1 but not allocating it. I added a few commits that allocates non-TLS compartment stacks.

**Comment by ayrtonm** [comment]
Also pad-tls and libc handling is still an issue but it's masked in CI because QEMU normally doesn't load libc.so into the address space anyway.

**Comment by ayrtonm** [comment]
Managed to get minimal test using criterion built for ARM running in QEMU. For some reason QEMU works in the github CI runner but I tried a separate ubuntu setup and QEMU does not seem to work. Looks like some issue with ld.so not applying relocations leading to a signal 7 bus error so I'll stick to testing in CI. Also in some test cases the test seemed to run (i.e. print to stdout) but the callgates.so wasn't loaded so we should have some debug print statement in the call gates for now to be sure we know what control-flow looks like.

**Comment by ayrtonm** [comment]
Need to drop the commit installing criterion in CI and this should work once #337 gets merged

**Comment by ayrtonm** [comment]
Turns out gcc doesn't support naked fns on arm64 so I'll have to rework some of these a bit to get the prologues right.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77882

**Review by rinon — APPROVED** [review]
LGTM after adding comments to both assembly functions.

File: runtime/libia2/include/ia2_internal.h
Comment (ayrtonm) [reply runtime/libia2/include/ia2_internal.h:214]: ah yeah I should've added the comments back in. I originally had some but I ran into the issue where missing a '\n' at the end of a comment line would silently comment out the instruction on the next line so I removed them while working on it. I'll try to refactor this macro and add them back in.

Comment (ayrtonm) [reply runtime/libia2/include/ia2_internal.h:214]: done

**Pull request merged**
